src/app/shared/components/header/header.component.ts

```diff
@@ -0,0 +1,46 @@
+import { Router } from '@angular/router';
+// import { ChangeDetectionStrategy } from '@angular/core';
+import { SITEURLS } from '@shared/siteUrls';
+import { AuthService } from '@shared/services/auth.service';
+import { Component } from '@angular/core';
+import { userObject } from '@shared/interfaces/userObject';
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.scss'],
+  // changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class HeaderComponent  {
+
+
+  constructor(
+    private authService: AuthService,
+    private router: Router
+    ) {}
+   
+  isLoggedIn: boolean;
+  siteUrls:object = SITEURLS;
+  userDetails:userObject
+
+  checkIfLoggedIn():void {
+    this.authService.currentLoginStatus.subscribe (
```

Comment:
it can be checked by the guard of the main component and userDetails can be passed to the header